drivers/mdio/mdio_nxp_enet.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_enet_mdio
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/mdio.h>
+#include <zephyr/drivers/ethernet/eth_nxp_enet.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/sys_clock.h>
+#include <soc.h>
+
+/* Target MDC frequency 2.5 kHz */
```

Comment:
looks like this is a typo, this is calculating MHz,

if interested here is a graph of the register value (5 bits 0-32) compared to the internal enet clock frequency used for mdc when using this calculation:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/110566814/d88738a1-787b-4b65-9038-717e99eb887a)

it always gives a value of at least 1 for the register

and corresponding graph of resulting mdc frequency:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/110566814/5e87cdac-e05e-406d-8b4e-1da29af143b5)

This equation comes from the SDK. I didn't try to come to a more optimal way of doing this, since the register value doesn't simply set a frequency but is relative to another clock, it is a bit complicated and I think the SDK probably spent a lot of time on this already. But it seems true that this will never result in the full 2.5 MHz for the MDC

Reply:
thanks for the details. Yes I understand that not always will be able to use the maximum frequency, but the target frequency caught being 2.5 KHz caught my eyes. If it's a typo in your code comments can it be fixed then?

Reply:
yes I fixed it